dask/array/tests/test_overlap.py

```diff
@@ -702,3 +723,55 @@ def test_trim_boundry(boundary):
     x_overlaped = da.overlap.overlap(x, 2, boundary=boundary)
     x_trimmed = da.overlap.trim_overlap(x_overlaped, 2, boundary=boundary)
     assert np.all(x == x_trimmed)
+
+
+def test_map_overlap_rechunks_array_if_needed():
+    # https://github.com/dask/dask/issues/6597
+    expected = np.arange(11)
+    x = da.from_array(expected, chunks=5)
+    y = x.map_overlap(lambda x: x, depth=2, boundary=0)
+    assert y.chunks == ((5, 6),)
+    assert_array_equal(y, expected)
+
+
+def test_map_overlap_rechunks_array_if_needed():
+    # https://github.com/dask/dask/issues/6597
+    expected = np.arange(11)
+    x = da.from_array(expected, chunks=2)
+    y = x.map_overlap(lambda x: x, depth=3, boundary=0)
+    assert y.chunks == ((4, 4, 3),)
+    assert_array_equal(y, expected)
+
+
+def test_map_overlap_rechunks_array_along_multiple_dims_if_needed():
+    # https://github.com/dask/dask/issues/6688
+    rand = da.random.random((860, 1024, 1024), chunks=(1, 1024, 1024))
+    filtered = rand.map_overlap(
+        lambda arr: arr,
+        depth=(2, 2, 2),
+        boundary="reflect",
+    )
+    assert filtered.chunks == ((2,) * 430, (1024,), (1024,))
+
+
+@pytest.mark.parametrize(
+    "chunks,expected",
+    [
+        [(10,), (10,)],
+        [(10, 10), (10, 10)],
+        [(20, 20, 20, 1), (20, 20, 11, 10)],
+        [(20, 20, 10, 1), (20, 20, 11)],
+        [(2, 20, 2, 20), (14, 10, 20)],
+        [(1, 1, 1, 1, 7), (11,)],
+        [(20, 20, 2, 20, 20, 2), (20, 12, 10, 20, 12, 10)],
+    ],
```

Comment:
I would suggest adding a case similar to `test_depth_greater_than_smallest_chunk_combines_chunks`, where concatenation does have to happen, and it creates chunks larger than any in the original, oi.e.
(10, 10, 1) -> (10, 11)

Reply:
good idea!